src/types/utils.cpp

```diff
@@ -378,71 +112,326 @@ til::color Utils::ColorFromHexString(const std::string_view str)
 }
 
 // Routine Description:
-// - Shorthand check if a handle value is null or invalid.
+// - Given a color string, attempts to parse the color.
+//   The color are specified by name or RGB specification as per XParseColor.
 // Arguments:
-// - Handle
+// - string - The string containing the color spec string to parse.
 // Return Value:
-// - True if non zero and not set to invalid magic value. False otherwise.
-bool Utils::IsValidHandle(const HANDLE handle) noexcept
+// - An optional color which contains value if a color was successfully parsed
+std::optional<til::color> Utils::ColorFromXTermColor(const std::wstring_view string) noexcept
 {
-    return handle != nullptr && handle != INVALID_HANDLE_VALUE;
+    std::optional<til::color> color = ColorFromXParseColorSpec(string);

```

Comment:
i'd recommend against it, only because it runs _both_ expensive parsing steps and then chooses the first one that succeeds.

coalesce isn't good for lazy evaluation.